src/library/basetracktablemodel.cpp

```diff
@@ -157,8 +160,23 @@ BaseTrackTableModel::BaseTrackTableModel(
 void BaseTrackTableModel::initTableColumnsAndHeaderProperties(
         const QStringList& tableColumns) {
     m_columnCache.setColumns(tableColumns);
-    if (m_columnHeaders.size() < tableColumns.size()) {
-        m_columnHeaders.resize(tableColumns.size());
+    DEBUG_ASSERT(m_columnHeaders.isEmpty());
+    m_columnHeaders.resize(tableColumns.size());
+    // Init the mapping of all columns, even for internal columns that are
+    // hidden/invisible. Otherwise mapColumn() would not return a valid result
+    // for those columns.
+    for (auto columnValue = 0; columnValue < ColumnCache::NUM_COLUMNS; ++columnValue) {
+        const auto column = static_cast<ColumnCache::Column>(columnValue);
+        DEBUG_ASSERT(column != ColumnCache::COLUMN_LIBRARYTABLE_INVALID);
+        const auto headerIndex = m_columnCache.fieldIndex(column);
+        if (headerIndex < 0) {
+            // Missing table column.
+            continue;
+        }
+        DEBUG_ASSERT(headerIndex < m_columnHeaders.size());
+        DEBUG_ASSERT(mapColumn(headerIndex) == ColumnCache::COLUMN_LIBRARYTABLE_INVALID);
```

Comment:
this doesn't look right, it shouldn't be invalid, right?
```suggestion
        DEBUG_ASSERT(mapColumn(headerIndex) != ColumnCache::COLUMN_LIBRARYTABLE_INVALID);
```

Reply:
before: invalid, after: valid

Reply:
Obsolete: https://github.com/mixxxdj/mixxx/pull/13682/commits/bb0495e7f1cfb2fc3fd327b90f8bc8a52ff79159

Coincidentally this just happened while switching the view during a library rescan. Never occurred before.